exts/ferrocene_spec/items_with_rubric.py

```diff
@@ -40,11 +40,7 @@ class RubricCollector(EnvironmentCollector):
     def clear_doc(self, app, env, docname):
         storage = get_storage(env)
         for rubric, items in storage.items():
-            # This makes a copy of the list (with `list(items)`) to be able to
-            # remove items from it without affecting the iteration.
-            for i, item in enumerate(list(items)):
-                if item.document == docname:
-                    items.pop(i)
+            items[:] = (item for item in items if item.document != docname)
```

Comment:
what's the advantage of doing it this way, instead of...

    items = [item for item in items if item.document != docname]

Reply:
(answering self) it modifies the list instad of creating the new one

Reply:
I suppose the generator is coerced to a list as well (but I don't know the right Python terminology), while the choice of generator is for being gentle to memory, where the items are created only when needed